Makefile

```diff
@@ -135,6 +135,14 @@ lintlua lintsh lintpy lintuncrustify lintc lintcfull check-single-includes gener
 
 test: functionaltest unittest
 
+iwyu: build/.ran-cmake
+	cmake --preset iwyu
+	cmake --build --preset iwyu > iwyu.log
+	iwyu-fix-includes --only_re="src/nvim" --ignore_re="src/nvim/(auto|map.h)" < iwyu.log
+	# IWYU adds an anonymous struct to map.h which gives an error when
+	# building. Skip fixing map.h for now until we figure out why this
+	# happens.
+
```

Comment:
```suggestion
```

this will not work if the project is already built, and alway re-building is expensive, furthermore, you're overwriting the log, 

so I'm not sure if this make target is that robust currently